### DISCUSSION TITLE
Built-in support for high availability

### DISCUSSION THREAD

**Discussion by nblumhardt** [datalust/seq-tickets#1103] [Ideas]
On the road to a highly-available Seq, we expect the next major release will add a two-node replicated configuration that will improve failure tolerance and avoid downtime in more scenarios.

RFC is TBD.

**Comment by nblumhardt** [comment]
Changing the title of this to better reflect where we're aiming with this feature.

**Comment by CrossBound** [comment]
Curious if you have an ETA when this will be available in the preview build... Also, will this be a breaking change or should I be able to upgrade an existing container/pod?

**Comment by nblumhardt** [comment]
Hi @CrossBound - thanks for jumping in. We plan to support a clean migration from existing Seq instances; we're targeting April for a preview build, we'll update this ticket with more information soon (we're also intending to publish details of the feature in an RFC before the preview).

**Comment by nblumhardt** [comment]
Removing this from the 2020.1 milestone. We'd originally expected a full native high-availability configuration to be supported with 2020, unfortunately this is not ready for public consumption, and despite a lot of implementation work being completed, the feature won't make it into the next version.

**Comment by CrossBound** [comment]
Looks like this hasn't made it into 2020.2 either, do you have any kind of ETA? This is a very important feature for us. We're not quite to the point of it being a requirement but we are intending to move to a kubernetes environment with Seq running as a pod and when we get there it will be important to have reliable HA on our logging infrastructure.

**Comment by nblumhardt** [comment]
Hi @CrossBound - thanks for the bump. We're still making forward progress on this - it's hard to determine when this will land, unfortunately, but we hope to have a beta of the first phase of it (two-node disaster-recovery configuration) sometime within the next few (six week) milestones.

**Comment by nblumhardt** [comment]
We've decided to move this ticket to a discussion so that we can cover the full range of HA and clustering requirements here.

We've made huge progress towards this in the the past year and will be releasing some small preview pieces soon, starting with #1102 .
